symengine/rational.cpp

```diff
@@ -78,10 +78,16 @@ bool Rational::__eq__(const Basic &o) const
 
 int Rational::compare(const Basic &o) const
 {
-    SYMENGINE_ASSERT(is_a<Rational>(o))
-    const Rational &s = static_cast<const Rational &>(o);
-    if (i == s.i) return 0;
-    return i < s.i ? -1 : 1;
+    if (is_a<Rational>(o)) {
+        const Rational &s = static_cast<const Rational &>(o);
+        if (i == s.i) return 0;
+        return i < s.i ? -1 : 1;
+    }
+    if (is_a<Integer>(o)) {
+        const Integer &s = static_cast<const Integer &>(o);
+        return mpq_cmp_z(i.get_mpq_t(), s.as_mpz().get_mpz_t());
```

Comment:
Arithmetic operators work for different combinations of `mpz_class` and mpq_class` right?